packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/read.rs

```diff
@@ -21,7 +28,52 @@ impl<C: AsClient> crud::Read<PersistedEntity> for PostgresStore<C> {
         &self,
         filter: &'f Self::Query<'q>,
     ) -> Result<Vec<PersistedEntity>, QueryError> {
-        let mut compiler = SelectCompiler::with_default_selection();
+        let mut compiler = SelectCompiler::new();
+
+        compiler.add_selection_path(
+            &EntityQueryPath::Properties(None),
+            Distinctness::Indistinct,
+            None,
+        );
+        compiler.add_selection_path(
+            &EntityQueryPath::Id,
+            Distinctness::Distinct,
+            Some(Ordering::Ascending),
```

Comment:
I think we might want to go to `None` here until we're ready to expose ordering through queries. Unless we're using this to establish `is_latest` or something then maybe we should get rid of it.

It's not going to be super cheap to do for every query

Reply:
I removed the ordering in cd24667145277c2de801d88e42dc5a90bce31ed1. I also made this function more readable by adding another add-selection method and default to `Indistinct` without ordering in d8d8c073bd58c2bfa24f97aa07b191a6b28a7dce

Reply:
@thehabbos007 I'm not 100% sure about the `RowIndex` change, I _think_ it looks good (I'm a very big fan of the impact on readability). 

Any chance you could have a look too?

Reply:
I agree that this is a great addition. I've previously asked if the selection ordering is stable, which Tim said it was. Because of that I think it's safe to return the indices up front. 

The trait itself is very simple, as it just checks the index is within range of the number of returned columns. Looks good to me :)